### ISSUE TITLE
[TSQL] parsing error for computed columns expression

### ISSUE THREAD

**Issue by tkachenkomaria244** [sqlfluff/sqlfluff#4971]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parsing error occurs for code with 
`create table dim
(
         [date] date primary key
	, [day] as datepart(day, [date])
	, [month] as datepart(month, [date])
	, [week] as datepart(iso_week, [date])
	, [dayofweek] as datepart(weekday, [date])
	, firstofweek as convert(date, dateadd(week, datediff(week, 0, [date]), 0)) 
	, [quarter] as datepart(quarter, [date])
	, [year] as datepart(year, [date])
	, firstofyear as convert(date, dateadd(year, datediff(year, 0, [date]), 0))
)`

### Expected Behaviour

No parsing error for this expected construction:
[CREATE TABLE (Transact-SQL) - SQL Server - column_name-as-computed_column_expression](https://learn.microsoft.com/en-us/sql/t-sql/statements/create-table-transact-sql?view=sql-server-ver16#column_name-as-computed_column_expression)

### Observed Behaviour

`==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'create table
                       | dim\n(\n\t[date] date primary ...'`

### How to reproduce

run `sqlfluff parse --dialect tsql 'script_for_sqlfluff.sql'`
script_for_sqlfluff.sql:
`create table dim
(
	[date] date primary key
	, [day] as datepart(day, [date])
	, [month] as datepart(month, [date])
	, [week] as datepart(iso_week, [date])
	, [dayofweek] as datepart(weekday, [date])
	, firstofweek as convert(date, dateadd(week, datediff(week, 0, [date]), 0)) 
	, [quarter] as datepart(quarter, [date])
	, [year] as datepart(year, [date])
	, firstofyear as convert(date, dateadd(year, datediff(year, 0, [date]), 0))
)`

### Dialect

tsql

### Version

sqlfluff 2.1.2

### Configuration

`[sqlfluff]
sql_file_exts = .sql
dialect = tsql
templater = placeholder
large_file_skip_byte_limit = 50000
max_line_length = 250`

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by skadyan** [comment]
The following is a condensed version of the SQL code that can be used to reproduce the issue
```
SELECT MONTH('2023-07-23' ) -- 7  It Works
;
SELECT DAY('2023-07-23' ) -- 23  It is failing but it shouldn't
;
```

**Comment by fmms** [comment]
@skadyan looking at your minimal example, this might be solved by #5062 ?

**Comment by fmms** [comment]
I hoped that this is now solved with #5062 merged. However the problem seems to be that no calculated columns are supported at all.

Here the statement in formatted so that it can be read easily:
```sql
CREATE TABLE dim (
  [date] date PRIMARY KEY
  , [day] as datepart(MONTH, [date])
  , [month] as datepart(month, [date])
  , [week] as datepart(iso_week, [date])
  , [dayofweek] as datepart(weekday, [date])
  , firstofweek as convert(date, dateadd(week, datediff(week, 0, [date]), 0))
  , [quarter] as datepart(quarter, [date]) , [year] as datepart(year, [date])
  , firstofyear as convert(date, dateadd(year, datediff(year, 0, [date]), 0))
)

```

And yes this is valid tsql.

**Comment by skadyan** [comment]
`cat sample.sql`
SELECT DAY('2023-07-23')

`sqlfluff fix sample.sql --dialect tsql  -v`
==== sqlfluff ====
sqlfluff:                2.3.0 python:                 3.10.9
implementation:        cpython verbosity:                   1
dialect:                  tsql templater:               jinja
rules:                                all
==== finding fixable violations ====
== [sample.sql] PASS
  [1 templating/parsing errors found]
==== no fixable linting violations found ====
All Finished 📜 🎉!

It does not seem to be working for me. Still getting parsing error.

**Comment by tkachenkomaria244** [comment]
It seems that some of datepart names and it's abbreviations are not parsed correctly.
[datepart-transact-sql](https://learn.microsoft.com/en-us/sql/t-sql/functions/datepart-transact-sql?view=sql-server-ver16).
For example, code parsing with tsql (sqlfluff parse script_for_sqlfluff.sql --dialect tsql)
```select
	datename(dw, evbw.period) as day_of_week
	, datepart(iso_week, evbw.period) as week_num
```
gives 'dw' as "date_part" segment, and 'iso_week' as "expression->column_reference->naked_identifier"
